### PR TITLE
Move all kotlin-client samples in subfolder kotlin

### PR DESCRIPTION
**Pull request by Ch4rl3x**
[Pre PR for Merging #5697](https://github.com/OpenAPITools/openapi-generator/pull/5697#issuecomment-604392117)

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### PR Description
This PR moves all kotlin-client sample projects into subfolder named 'kotlin' for better merging and less diffs in PR #5697

@jimschubert @dr4ke616 @karismann @Zomzog @andrewemery @4brunu

### CODE REVIEW

File: samples/client/petstore/kotlin/retrofit2/src/main/kotlin/org/openapitools/client/auth/OAuthOkHttpClient.kt [hunk 1]
```diff
@@ -0,0 +1,61 @@
+package org.openapitools.client.auth
```
Comment (4brunu) [inline samples/client/petstore/kotlin/retrofit2/src/main/kotlin/org/openapitools/client/auth/OAuthOkHttpClient.kt]: I think there are some files like this that don't belong to this PR.
Because the OAuth supports is still pending in the PR https://github.com/OpenAPITools/openapi-generator/pull/5697
Can you please generate the sample projects again?

File: samples/client/petstore/kotlin/retrofit2/src/main/kotlin/org/openapitools/client/apis/UserApi.kt [hunk 2]
```diff
@@ -10,28 +10,28 @@ import okhttp3.MultipartBody
 import org.openapitools.client.models.User
 
 interface UserApi {
-    @POST("/user")
```
Comment (4brunu) [inline samples/client/petstore/kotlin/retrofit2/src/main/kotlin/org/openapitools/client/apis/UserApi.kt]: I'm not sure if this change is right or not...
Can someone else give some feedback here?

File: bin/windows/kotlin-client-petstore.bat [hunk 3]
```diff
@@ -5,6 +5,6 @@ If Not Exist %executable% (
 )
 
 REM set JAVA_OPTS=%JAVA_OPTS% -Xmx1024M -DloggerPath=conf/log4j.properties
-set ags=generate  --artifact-id "kotlin-petstore-client" -i modules\openapi-generator\src\test\resources\2_0\petstore.yaml -g kotlin -o samples\client\petstore\kotlin --additional-properties dateLibrary=java8,serializableModel=true
```
[the remark this answers is not in the document]
Comment (Ch4rl3x) [reply bin/windows/kotlin-client-petstore.bat]: Hmm... :) No reasons. Modified the files more than once. Maybe i didnt changed it back. For these PR, i copied the modified bat and sh files from PR #5697 without changing them

[the remark this answers is not in the document]
Comment (4brunu) [reply bin/windows/kotlin-client-petstore.bat]: ok, thanks :)

File: samples/client/petstore/kotlin/retrofit2/src/main/kotlin/org/openapitools/client/auth/OAuthOkHttpClient.kt (hunk 1 above)
Comment (Ch4rl3x) [reply samples/client/petstore/kotlin/retrofit2/src/main/kotlin/org/openapitools/client/auth/OAuthOkHttpClient.kt]: I forgott to mvn clean install before rerun generate all sample projects. Fixed with https://github.com/OpenAPITools/openapi-generator/pull/5718/commits/a9a01a4d01ba6366569ce82021773e8abb936da3

**Comment by 4brunu** [comment]
Looks good to me. 👍 
Lets see what the other members of the technical committee think about this and if the CI passes.

File: samples/client/petstore/kotlin/retrofit2/src/main/kotlin/org/openapitools/client/apis/UserApi.kt (hunk 2 above)
Comment (Ch4rl3x) [reply samples/client/petstore/kotlin/retrofit2/src/main/kotlin/org/openapitools/client/apis/UserApi.kt]: There is still a bug in kotlin-retrofit2 generator. These releative pathes have to start WITHOUT slashes. See java-retrofit2 generator. I fixed it within my PR #5697 
https://github.com/OpenAPITools/openapi-generator/pull/5697/commits/aaef017a5a730812ac3c209cb106edda4f4349cd#diff-b60a03ff3897718b90d1c4d54850d0de in file KotlinClientCodegen.java

Comment (4brunu) [reply samples/client/petstore/kotlin/retrofit2/src/main/kotlin/org/openapitools/client/apis/UserApi.kt]: Can you elaborate a bit more on why that's a bug?
I thought that "/user" was the correct behaviour.

Comment (Ch4rl3x) [reply samples/client/petstore/kotlin/retrofit2/src/main/kotlin/org/openapitools/client/apis/UserApi.kt]: /user was the behaviour before my PR #5697. Retrofit2 does not allow a baseURL without trailing slash. That means, the relative path has to start without slash. The current version of kotlin-retrofit2 didnt worked in our projekt. We checked the java-retrofit2 templates. The relative pathes in the api files in java-retrofit2 didnt start with slash.

We ran into following issue: https://github.com/square/retrofit/issues/907

**Comment by wing328** [comment]
I think you will also need to update https://github.com/OpenAPITools/openapi-generator/blob/a9a01a4d01ba6366569ce82021773e8abb936da3/pom.xml#L1451-L1461

**Comment by Ch4rl3x** [comment]
> I think you will also need to update
> 
> https://github.com/OpenAPITools/openapi-generator/blob/a9a01a4d01ba6366569ce82021773e8abb936da3/pom.xml#L1451-L1461

Should I edit the file manually or is it generated by a shell script

**Comment by wing328** [comment]
That file is manually maintained

**Comment by Ch4rl3x** [comment]
I updated the pom.xml in root path, but still errors in Shippable Ci. Missing pom.xml files in all kotlin projects. Anything to todo?

**Comment by wing328** [comment]
We manually added a (unique) pom.xml file in the Kotlin sample folder. You will need to move the pom.xml to the new directory in order for maven to run the integration tests

**Comment by Ch4rl3x** [comment]
Anything to do? Don't know why these ci builds failed

**Comment by 4brunu** [comment]
I don't know what's the problem, but looking at the travis config file, 
https://github.com/OpenAPITools/openapi-generator/blob/master/.travis.yml#L144
It points to 
https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator-maven-plugin/examples/kotlin.xml

Maybe something needs to be updated there?
I'm not sure.
@wing328 do you have any tips on how to fix travis?
